src/backend/test.rs

```diff
@@ -343,6 +365,75 @@ impl Backend for TestBackend {
     fn flush(&mut self) -> io::Result<()> {
         Ok(())
     }
+
+    #[cfg(feature = "scrolling-regions")]
+    fn scroll_region_up(&mut self, region: std::ops::Range<u16>, scroll_by: u16) -> io::Result<()> {
+        let width: usize = self.buffer.area.width.into();
+        let region =
+            region.start.min(self.buffer.area.height)..region.end.min(self.buffer.area.height);
```

Comment:
nit: Keeping this in a single region variable over using a start and end variable seems like it hampers the readability a little.

Reply:
Okay, I broke it apart.